app/controllers/v0/sessions_controller.rb

```diff
@@ -82,7 +82,7 @@ def saml_user
 
     def create_saml_user
       user = User.new(user_attributes)
-      user = Decorators::MviUserDecorator.new(user).create unless user.loa1? || user.gender.nil?
+      user.mvi # caches mvi data
```

Comment:
@kreek instead of in the user initializer, you could maybe create a method that memoizes when invoked, otherwise returns nil and it would continue to make request until value is not nil.

Possibly consider making it a private method on user, and the decorated methods on user.rb could just invoke it when required.

```ruby
def mvi
  @mvi ||= call_to_fetch_mvi rescue nil
end
```